### PR TITLE
Add version naming and deprecation link

### PR DESCRIPTION
**Pull request by dtzar**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Comment by dtzar** [comment]
Still waiting on a review here.  PR is still valid.

File: documentation/MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Guide for migrating to `sdk/resourcemanager/**/arm**` from `services/**/mgmt/**`
 
-This document is intended for users that are familiar with the previous version of the Azure SDK For Go for management modules (`services/**/mgmt/**`) and wish to migrate their application to the next version of Azure resource management libraries (`sdk/resourcemanager/**/arm**`)
+This document is intended for users that are familiar with the previous version of the Azure SDK For Go for management modules (`services/**/mgmt/**` also known as track or version 1) and wish to migrate their application to the next version of Azure resource management libraries (`sdk/resourcemanager/**/arm**` also known as track or version 2).  All version 1 Go modules are [deprecated and no longer recieving support](https://azure.github.io/azure-sdk/releases/deprecated/go.html).
```
Comment (josefree) [inline documentation/MIGRATION_GUIDE.md:3]: This document is intended for users that are familiar with the older version of Azure SDKs of management libraries for Go language (package name starts with `services/**/mgmt/**` ) and pursue to migrate their application code on to the latest version of Azure SDKs for Go that conform to our [current Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html) (SDK package name starts with `sdk/resourcemanager/**/arm**` ).  The older Golang libraries that do not conform to [Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html) are deprecated by 30 September 2023.  Although the older libraries can still be used beyond deprecation, they will no longer receive official support and updates from Microsoft after that date.

**Review by tadelesh — APPROVED** [review]

**Comment by dtzar** [comment]
Thanks for merging @tadelesh.  Can you also please go ahead and merge https://github.com/Azure/azure-sdk-for-go/pull/21680 since it completes the visibility for the same topic?

**Comment by tadelesh** [comment]
@dtzar I've merged the PR. Thanks for your contribution.

**Pull request merged**